sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala

```diff
@@ -625,6 +625,21 @@ class QueryExecutionErrorsSuite
     }
   }
 
+  test("BINARY_ARITHMETIC_CAUSE_OVERFLOW: byte plus byte result overflow") {
+    withSQLConf("spark.sql.ansi.enabled" -> "true") {
+      checkError(
+        exception = intercept[SparkArithmeticException] {
+          sql(s"select CAST('127' AS TINYINT) + CAST('5' AS TINYINT)").collect()
```

Comment:
You can create a TINYINT constant directly without casting:
```suggestion
          sql(s"select 127Y + 5Y").collect()
```

Reply:
Thanks for your reminder. Done